fastchat/conversation.py

```diff
@@ -43,6 +43,7 @@ class Conversation:
     # The names of two roles
     roles: Tuple[str] = ("USER", "ASSISTANT")
     # All messages. Each item is (role, message).
+    # Each message is either a string or a tuple of (string, PIL.Image).
```

Comment:
Is this format compatible with GPT-4 Vision API?
We have to make sure we can do round-trip conversion between our format and GPT-4 vision API format.

As a test case, please write an example similar to this one https://github.com/lm-sys/FastChat/blob/e46d97a368688554c7f1dfa2525d21dd86117fb5/fastchat/conversation.py#L1218-L1224